mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h

```diff
@@ -6986,110 +7037,246 @@ struct GridwiseGemmV2RewritePattern
-                                                b.getStringAttr("m1"),
-                                                b.getStringAttr("m2")})),
-                 b.getNamedAttr("lower_layer_dimensions",
-                                b.getArrayAttr({b.getI32IntegerAttr(1)})),
-                 b.getNamedAttr("lower_layer_names",
-                                b.getArrayAttr({b.getStringAttr("gemmM")})),
-                 b.getNamedAttr("transformation", b.getStringAttr("Embed")),
-                 b.getNamedAttr("parameters",
-                                b.getArrayAttr({b.getI32IntegerAttr(M1 * M2),
-                                                b.getI32IntegerAttr(M2),
-                                                b.getI32IntegerAttr(1)})),
-             }),
-             b.getDictionaryAttr(
-                 {b.getNamedAttr("upper_layer_dimensions",
-                                 b.getArrayAttr({b.getI32IntegerAttr(4)})),
-                  b.getNamedAttr("upper_layer_names",
-                                 b.getArrayAttr({b.getStringAttr("n")})),
-                  b.getNamedAttr("lower_layer_dimensions",
-                                 b.getArrayAttr({b.getI32IntegerAttr(2)})),
-                  b.getNamedAttr("lower_layer_names",
-                                 b.getArrayAttr({b.getStringAttr("gemmN")})),
-                  b.getNamedAttr("transformation",
-                                 b.getStringAttr("PassThrough"))})})));
-    // set lower_layer_layout attribute.
-    transformedNewOutputAttrs.push_back(b.getNamedAttr(
-        "lower_layer_layout",
-        b.getArrayAttr({b.getStringAttr("gemmG"), b.getStringAttr("gemmM"),
-                        b.getStringAttr("gemmN")})));
-    // set upper_layer_layout attribute.
-    transformedNewOutputAttrs.push_back(b.getNamedAttr(
-        "upper_layer_layout",
-        b.getArrayAttr({b.getStringAttr("g"), b.getStringAttr("m0"),
-                        b.getStringAttr("m1"), b.getStringAttr("m2"),
-                        b.getStringAttr("n")})));
-    auto newOutputType = MemRefType::get(
-        {G, M0, M1, M2, N}, outputType.getElementType(), newOutputAffineMaps);
-    auto newOutputTransformOp = b.create<miopen::TransformOp>(
-        loc, newOutputType, op.output(), transformedNewOutputAttrs,
-        /*populateBounds=*/true);
-
-    // Build affine expression for Sequence<1, M0, 1, M2, 1>
-    // (d0, d1, d2, d3, d4) -> (d1 * M2 + d3)
-    auto matrixCAffineMap5to1 = AffineMap::get(
-        5, 0,
-        {getAffineDimExpr(1, ctx) * getAffineConstantExpr(M2, ctx) +
-         getAffineDimExpr(3, ctx)},
-        op.getContext());
+    auto *ctx = op.getContext();
 
+    Value newOutputTransformOp;
+    SmallVector<Value, 4> vectors;
+    vectors.reserve(tailResults.size());
+    AffineMap tensorToXdlopsVectorMap;
+    if (enableOutSwizzles) {
+      // The swizzle operation doesn't fundamentally affect the mapping
```

Comment:
For the attribute maps here, perhaps it'd be a good idea to have a meeting with @asroy / @zjing14 together so we review the logic together?

Reply:
Not a bad idea, can you schedule?